evm_loader/program/src/evm/opcode.rs

```diff
@@ -616,7 +616,8 @@ impl<B: Database> Machine<B> {
     pub fn opcode_blockhash(&mut self, backend: &mut B) -> Result<Action> {
         let block_hash = {
             let block_number = self.stack.pop_u256()?;
-            backend.block_hash(block_number)?
+            // geth converts to u64, ok lets do the same
+            backend.block_hash(block_number.as_u64())?
```

Comment:
Geth
https://github.com/ethereum/go-ethereum/blob/0fb1be09303ef509d7824f0b9415a3645149437a/core/vm/instructions.go#L433
```go
	num := scope.Stack.peek()
	num64, overflow := num.Uint64WithOverflow()
	if overflow {
		num.Clear()
		return nil, nil
	}
```

If number > u64::max, Geth pushes zero to stack.
You silently ignore it.